HHS/Head-Start-TTADP
# [TTAHUB-3097] Implement transactions for worker processes

## Description of change
This update introduces transaction handling for worker processes in the Head Start TTA Data Platform. The main changes include:

1. Implementing transaction management in worker scripts to ensure data consistency.
2. Enhancing error logging for detailed information about errors and database connection states.
3. Updating tests to align with the new transaction management and error handling.


## How to test

1. Run Unit Tests: Execute all unit tests to ensure they pass with the new transaction handling logic.
2. Simulate Worker Processes: Trigger worker processes and monitor logs for detailed error messages.
3. Check Database Consistency: Verify that data remains consistent and correctly rolled back in case of errors during worker processes. 

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3097


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
